### PR TITLE
Some profiler refactoring

### PR DESCRIPTION
**Pull request by trask**
The primary goal of this PR is to make the profiler code structure clearer top-down (packages + public classes).

The proposed structure of packages + public classes in this PR is:

* ...profiler
  * ProfilingInitializer
  * Profiler
* ...profiler.config
  * ConfigPollingInit
  * ProfilerConfiguration
* ...profiler.service
  * ArtifactAcceptedResponse
  * BlobAccessPass
  * ServiceProfilerClient
* ...profiler.trigger
  * AlertingSubsystemInit
  * AlertTriggerSpanProcessor
* ...profiler.upload
  * ServiceProfilerIndex
  * UploadFinishArgs
  * UploadListener
  * UploadService

This also consolidates the above profiler classes from three modules into just the `agent-tooling` module, eliminating `agent-service-profiler` and `agent-profiler-api`.

The PR also does a few other things in the profiler code:
* Introduces `@AutoValue` (and builders) in several places where they are helpful
* Replaces Moshi json library usage with Jackson json library usage
* Rewrites service tests to use Azure SDK "playback" http responses

### CODE REVIEW

**Review by johnoliver — APPROVED** [review]

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/AlternativeJfrConfigurations.java [hunk 1]
```diff
@@ -16,8 +16,9 @@
 import org.slf4j.LoggerFactory;
 
 /** Allows loading alternative jfc configuration files. */
-public class AlternativeJfrConfigurations {
-  private static final Logger LOGGER = LoggerFactory.getLogger(AlternativeJfrConfigurations.class);
+class AlternativeJfrConfigurations {
+
+  private static final Logger logger = LoggerFactory.getLogger(AlternativeJfrConfigurations.class);
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/AlternativeJfrConfigurations.java:21]: ClientLogger?

File: agent/agent-profiler/agent-alerting/src/main/java/com/microsoft/applicationinsights/alerting/AlertingSubsystem.java [hunk 2]
```diff
@@ -30,6 +29,7 @@
  * necessary issue an alert.
  */
 public class AlertingSubsystem {
+
   private static final Logger LOGGER = LoggerFactory.getLogger(AlertingSubsystem.class);
```
Comment (heyams) [inline agent/agent-profiler/agent-alerting/src/main/java/com/microsoft/applicationinsights/alerting/AlertingSubsystem.java:33]: ClientLogger here?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/Profiler.java [hunk 3]
```diff
@@ -45,14 +43,15 @@
  *   <li>Creates profiles on demand
  * </ul>
  */
-public class JfrProfiler implements ProfilerConfigurationHandler, Profiler {
-  private static final Logger LOGGER = LoggerFactory.getLogger(JfrProfiler.class);
+public class Profiler {
+
+  private static final Logger logger = LoggerFactory.getLogger(Profiler.class);
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/Profiler.java:48]: should we move to clientLogger for Profiler everywhere?

**Review by heyams — APPROVED** [review]

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/AlternativeJfrConfigurations.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/AlternativeJfrConfigurations.java:21]: ya, probably, let's do this in a separate PR across the whole codebase

**Pull request merged**